static/app/views/dashboards/editAccessSelector.tsx

```diff
@@ -48,6 +57,18 @@ function EditAccessSelector({dashboard, onChangeEditAccess}: EditAccessSelectorP
   const [isMenuOpen, setMenuOpen] = useState<boolean>(false);
   const {teams: selectedTeam} = useTeamsById({ids: [selectedOptions[1]]});
 
+  // Effect to update selectedOptions whenever the dashboard changes
+  useEffect(() => {
+    const teamIdsList: string[] = Object.values(teamsToRender).map(team => team.id);
+    if (!defined(dashboard.permissions) || dashboard.permissions.isEditableByEveryone) {
+      setSelectedOptions(['_creator', '_allUsers', ...teamIdsList]);
+    } else {
+      const permittedTeamIds =
+        dashboard.permissions.teamsWithEditAccess?.map(teamId => String(teamId)) ?? [];
+      setSelectedOptions(['_creator', ...permittedTeamIds]);
+    }
+  }, [dashboard, teamsToRender]);
+
```

Comment:
I had to add in this `useEffect` because I was having the issue where a re-render was not updating the selected permissions. Mainly ran into this issue when I updated the edit access on the `DashboardDetails` page and then went back to the `ManageDashboards` page and the access column was not showing the updated access

Reply:
Makes sense, `selectedOptions` was getting updated only when opening/closing the menu, and when changing the selections. I'll probably refactor this soon so that the `getSelectedOptions()` function can use `useEffect` and depend on `dashboard`. This looks good for now!